### PR TITLE
Api version check

### PR DESCRIPTION
**Pull request by MattGoldmanSSW**

### CODE REVIEW

File: Xamarin/SSW.Consulting/SSW.Consulting/Helpers/ApiInfo.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+using Newtonsoft.Json.Linq;
+
+namespace SSW.Consulting.Helpers
+{
+    public class ApiInfo
+    {
+        private string _baseUrl { get; set; }
+        private HttpClient _client;
+
+        public ApiInfo(string BaseURL)
```
Comment (AndreasLengkeek) [inline Xamarin/SSW.Consulting/SSW.Consulting/Helpers/ApiInfo.cs]: Make sure method arguments are using camelCase as per C# naming conventions No. 2
SSW: https://rules.ssw.com.au/follow-naming-conventions
C#: https://github.com/ktaranov/naming-convention/blob/master/C%23%20Coding%20Standards%20and%20Naming%20Conventions.md

File: Xamarin/SSW.Consulting/SSW.Consulting/App.xaml.cs [hunk 2]
```diff
@@ -74,5 +76,15 @@ private async Task UpdateAccessTokenAsync()
...
         }
+
+        private async Task CheckApiCompatibilityAsync()
+        {
+            ApiInfo info = new ApiInfo(Constants.ApiBaseUrl);
+
+            bool compatible = await info.IsApiCompatibleAsync();
+
+            if (!compatible)
+                await Application.Current.MainPage.DisplayAlert("Update Required", "Looks like you're using an older version of the app. You can continue, but some features may not function as expected.", "OK");
```
Comment (AndreasLengkeek) [inline Xamarin/SSW.Consulting/SSW.Consulting/App.xaml.cs]: This is a different message from what we have in the PBI.
I do like this one better so maybe we should update the Acceptance Criteria - check with PO

File: Xamarin/SSW.Consulting/SSW.Consulting/App.xaml.cs [hunk 3]
```diff
@@ -74,5 +76,15 @@ private async Task UpdateAccessTokenAsync()
...
             }
         }
+
+        private async Task CheckApiCompatibilityAsync()
+        {
+            ApiInfo info = new ApiInfo(Constants.ApiBaseUrl);
+
+            bool compatible = await info.IsApiCompatibleAsync();
+
+            if (!compatible)
```
[the remark this answers is not in the document]
Comment (MattGoldmanSSW) [reply Xamarin/SSW.Consulting/SSW.Consulting/App.xaml.cs]: Change made as requested

File: Xamarin/SSW.Consulting/SSW.Consulting/App.xaml.cs (hunk 2 above)
Comment (MattGoldmanSSW) [reply Xamarin/SSW.Consulting/SSW.Consulting/App.xaml.cs]: Change made as requested

File: Xamarin/SSW.Consulting/SSW.Consulting/Helpers/ApiInfo.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            }
+        }
+
+        public async Task<bool> IsApiCompatibleAsync()
+        {
+            string appVer = Constants.MaxApiSupportedVersion;
+
+            string apiVer = await GetApiVersionAsync();
+
+            if (apiVer == appVer)
```
[the remark this answers is not in the document]
Comment (MattGoldmanSSW) [reply Xamarin/SSW.Consulting/SSW.Consulting/Helpers/ApiInfo.cs]: Change made as requested

**Pull request merged**